src/Core/Error.js

```diff
@@ -148,10 +163,15 @@ function catchErrorBoundary (element, exception, owner) {
  * @param {Exception} exception
  */
 function propagateErrorBoundary (element, host, parent, exception) {
-	clearErrorBoundary(parent)
-	catchErrorBoundary(parent, exception, parent.owner)
+	if (element === parent)
+		if (element !== host)
+			throw exception
+
+	if (exception.origin !== SharedComponentWillUnmount)
+		if (exception.exception !== exception)
```

Comment:
`exception.exception` confusing

Reply:
I guess it could be anything else, or maybe it could be a public API that custom renderer can use to explicitly prevent the clear mechanics, this is currently used by the server render to archive this.